src/ai-providers/codex-cli.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * src/ai-providers/codex-cli.js
+ *
+ * Codex CLI provider implementation using the ai-sdk-provider-codex-cli package.
+ * This provider uses the local OpenAI Codex CLI with OAuth (preferred) or
+ * an optional OPENAI_API_KEY if provided.
+ */
+
+import { createCodexCli } from 'ai-sdk-provider-codex-cli';
+import { BaseAIProvider } from './base-provider.js';
+import { execSync } from 'child_process';
+import { log } from '../../scripts/modules/utils.js';
+import { getCodexCliSettingsForCommand } from '../../scripts/modules/config-manager.js';
+
+let _codexCliChecked = false;
+let _codexCliAvailable = null;
+
+export class CodexCliProvider extends BaseAIProvider {
+	constructor() {
+		super();
+		this.name = 'Codex CLI';
+		// Codex CLI has native schema support, no explicit JSON schema mode required
+		this.needsExplicitJsonSchema = false;
+		// Restrict to supported models for OAuth subscription usage
+		this.supportedModels = ['gpt-5', 'gpt-5-codex'];
+	}
+
+	/**
+	 * Codex CLI does not require an API key when using OAuth via `codex login`.
+	 * @returns {boolean}
+	 */
+	isRequiredApiKey() {
+		return false;
+	}
+
+	/**
+	 * Returns the environment variable name used when an API key is provided.
+	 * Even though the API key is optional for Codex CLI (OAuth-first),
+	 * downstream resolution expects a non-throwing implementation.
+	 * @returns {string}
+	 */
+	getRequiredApiKeyName() {
+		return 'OPENAI_API_KEY';
+	}
```

Comment:
are we sure we want to make it "openai_api_key" ? I would rather distinguish it like we did with grok-cli.js, with a separate env variable name for it